CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesBySuffixNGTest.java

```diff
@@ -0,0 +1,316 @@
+        mergeTypeParameter.setBooleanValue(false);
+        parameters.addParameter(mergeTypeParameter);
+
+        final PluginParameter<IntegerParameterType.IntegerParameterValue> thresholdParameter = IntegerParameterType.build(THRESHOLD_PARAMETER_ID);
+        thresholdParameter.setIntegerValue(0);
+        parameters.addParameter(thresholdParameter);
+
+        final PluginParameter<BooleanParameterType.BooleanParameterValue> mergeParameter = BooleanParameterType.build(MERGER_PARAMETER_ID);
+        mergeParameter.setBooleanValue(false);
+        parameters.addParameter(mergeParameter);
+
+        final PluginParameter<BooleanParameterType.BooleanParameterValue> leadParameter = BooleanParameterType.build(LEAD_PARAMETER_ID);
+        leadParameter.setBooleanValue(false);
+        parameters.addParameter(leadParameter);
+
+        final PluginParameter<BooleanParameterType.BooleanParameterValue> selectedParameter = BooleanParameterType.build(SELECTED_PARAMETER_ID);
+        selectedParameter.setBooleanValue(false);
+        parameters.addParameter(selectedParameter);
+
+        final Map<String, PluginParameter<?>> parametersMap = parameters.getParameters();
+
+        instance.updateParameters(parametersMap);
+
+        assertEquals(parametersMap.size(), 5);
+        assertTrue(parametersMap.get(MERGE_TYPE_PARAMETER_ID).isEnabled());
+        assertTrue(parametersMap.get(THRESHOLD_PARAMETER_ID).isEnabled());
+        assertEquals(parametersMap.get(THRESHOLD_PARAMETER_ID).getDescription(), "The suffix length to consider");
+        assertEquals(parametersMap.get(THRESHOLD_PARAMETER_ID).getIntegerValue(), 9);
+        assertTrue(parametersMap.get(MERGER_PARAMETER_ID).isEnabled());
+        assertTrue(parametersMap.get(LEAD_PARAMETER_ID).isEnabled());
+        assertTrue(parametersMap.get(SELECTED_PARAMETER_ID).isEnabled());
+    }
+
+    /**
+     * Test of getNodesToMerge method, of class MergeNodesBySuffix, with no
+     * identifier Attribute
+     */
+    @Test
+    public void testGetNodesToMerge_NoIdentifierAttribute() throws Exception {
+        System.out.println("testGetNodesToMerge_NoIdentifierAttribute");
+        Comparator<String> leadVertexChooser = null;
+        int threshold = 0;
+        boolean selectedOnly = false;
+        MergeNodesBySuffix instance = new MergeNodesBySuffix();
+        Map<Integer, Set<Integer>> expResult = new HashMap<>();
+
+        // create a new analytic graph
+        final Schema schema = SchemaFactoryUtilities.getSchemaFactory(AnalyticSchemaFactory.ANALYTIC_SCHEMA_ID).createSchema();
+        StoreGraph graph = new StoreGraph(schema);
+
+        // add vertices
+        int vxId1 = graph.addVertex();
+        int vxId2 = graph.addVertex();
+        int vxId3 = graph.addVertex();
+
+        Map<Integer, Set<Integer>> result = instance.getNodesToMerge(graph, leadVertexChooser, threshold, selectedOnly);
+        assertEquals(result, expResult);
+    }
+
+    //Hard to test without dialog box access
```

Comment:
Can you prefix this comment with a `TODO:` so its easier from the IDE to follow up on this later on?